### ISSUE TITLE
Yarn autoprunes local node_modules folder instead of the one specified with `yarn install --modules-folder`

### ISSUE THREAD

**Issue by wincent** [yarnpkg/yarn#6839]
As part of an Electron app build I run `yarn install --modules-folder $PROJECT_ROOT/build/MyApp.app/Contents/Resources/app/node_modules` to put the dependencies in the built app. Yarn then comes along and autoprunes *everything* inside `$PROJECT_ROOT/node_modules`, which breaks everything in the project repo. This is counter-intuitive. If I pass `--modules-folder`, I would expect Yarn to do its auto-pruning there alone.

My work around for now is to reinstall in the top-level after every build. Alternatively, I may see if I can copy the `yarn.lock` down into the build folder and execute from there instead.

Related:

* https://github.com/yarnpkg/yarn/issues/4462: "Is there a way to stop yarn from auto pruning after install?"
* https://github.com/yarnpkg/yarn/issues/5709: "yarn is deleting symlinks in node_modules that it doesn't own"
* https://github.com/yarnpkg/yarn/issues/5289: "Yarn install on a different folder using --modules-folder clears the default node_modules folder" — this one was closed as a duplicate (maybe incorrectly? — the other issue — https://github.com/yarnpkg/yarn/issues/1684: "How to use --modules-folder flag?" — seems quite a bit broader).

**Comment by rally25rs** [comment]
Yeah, this bug has unfortunately been open for a while. I found it myself once too but never found the time to dive in and fix it. Going to mark this as a duplicate of the bug I created a while ago.

If you are willing to make a PR to fix it, we would appreciate the help. It shouldn't be that hard... probably a hard-coded `node_modules` somewhere. I have some time off my regular job for the holidays, so I might be able to fix it next week if no one else tackles it...

duplicate #5419
